aten/src/ATen/native/README.md

```diff
@@ -64,6 +64,9 @@ signature.
   have ordinarily passed an `int` or `size_t`.
 - `double`. There is no `float`; ATen policy is to use `double` anywhere you would have used `float`.
 - `bool`
+- `Scalar`. `Scalar` supports binding to any numerical types from Python, including integeral types,
+  floating point types, and zero dimensional tensors. `int64_t` and `double` can only bind to the
+  corresponding Python numerical types.
```

Comment:
I think we should mention that you probably _don't_ want to use Scalar.  It's really used for binding to TH/THC code "real" types where the APIs you are binding to are actually different types.  double and int64_t arguments should suffice for most algorithms.

Reply:
Do we have a requirement that Scalar's scalar-type match the tensor type? (Actually, ISTR we *removed* the assert that each passed in Tensor had the same scalar type?)

Reply:
@gchanan I stole part of your comments above and put it into the new commit. :)

Reply:
@ezyang No, there's no requirement that the scalar types match (actually, that doesn't fully make sense because Scalars only hold doubles or int64_ts, but there are many more scalar types).  For the Tensor scalar type matching, that was only a Variable construct, native functions themselves never had that requirement.